pkgs/development/python-modules/funsor/default.nix

```diff
@@ -0,0 +1,88 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchFromGitHub
+, makefun
+, multipledispatch
+, numpy
+, opt-einsum
+, typing-extensions
+, pyro-ppl
+, torch
+, black
+, flake8
+, isort
+, pandas
+, pillow
+, pyro-api
+, pytest-xdist
+, pytestCheckHook
+, requests
+, scipy
+, torchvision
+}:
+
+let
+  funsor = buildPythonPackage rec {
+    pname = "funsor";
+    version = "0.4.5";
+
+    disabled = pythonOlder "3.7";
+
+    src = fetchFromGitHub {
+      owner = "pyro-ppl";
+      repo = pname;
+      rev = "refs/tags/${version}";
+      hash = "sha256-3JxnRlJN/kYstcIKt8cvDmqnp8oyoA4jkMBDVuAMa5Q=";
+    };
+
+    propagatedBuildInputs = [
+      makefun
+      multipledispatch
+      numpy
+      opt-einsum
+      typing-extensions
+    ];
+
+    nativeCheckInputs = [
+      # Backend
+      pyro-ppl
+      torch
+
+      black
+      flake8
+      isort
```

Comment:
```suggestion
```
those are linters and tests should not break if code quality is not correct

Reply:
Ok !